### PR TITLE
OSSM-81 Add 'streaming-elasticsearch' template

### PR DESCRIPTION
**Pull request by objectiser**
NOTE: This PR should not be merged until Jaeger 1.15 is available.

It is an almost exact copy of the `production-elasticsearch` template - with the strategy set to `streaming` and an extra spec element for `ingester` to enable the image and annotations to be configured.

The current "auto-provisioning" support for Kafka requires the user to install the strimzi/AMQ operator before Jaeger (and therefore Maistra). We need to decide whether the AMQ operator should be treated in a similar way to the Elasticsearch Operator, and be "[required](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md#required-crds)" (i.e. in the CSV)?

On one hand, the AMQ operator is not a "required" dependency - Jaeger could run in a production environment without it being present, by simply using elasticsearch storage.

However at the moment it is a convenient mechanism to reduce the work required by the end user to setup the Service Mesh.

### CODE REVIEW

**Comment by objectiser** [comment]
cc @jpkrohling - could you do a quick review to make sure all looks ok?

File: helm/istio/charts/tracing/templates/jaeger-streaming-elasticsearch.yaml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    annotations:
+      {{- range $key, $value := .Values.jaeger.annotations }}
+        {{ $key }}: {{ $value | quote }}
+      {{- end }}
+
+  ingester:
+    {{- if and .Values.jaeger.hub .Values.jaeger.tag .Values.jaeger.ingesterImage }}
+    image: {{ .Values.jaeger.hub }}/{{ .Values.jaeger.ingesterImage }}:{{ .Values.jaeger.tag }}
+    {{- end }}
+    annotations:
```
Comment (jpkrohling) [inline helm/istio/charts/tracing/templates/jaeger-streaming-elasticsearch.yaml:39]: Not quite sure how the template works, but don't you have to specify the broker URLs?

Comment (objectiser) [reply helm/istio/charts/tracing/templates/jaeger-streaming-elasticsearch.yaml:39]: No, as this will use the self provisioned approach.

Comment (jpkrohling) [reply helm/istio/charts/tracing/templates/jaeger-streaming-elasticsearch.yaml:39]: Sorry, I misread your initial comment and thought you were avoiding the auto provisioning feature. LGTM then

**Review by jpkrohling — APPROVED** [review]

**Review by rcernich — APPROVED** [review]

**Review by brian-avery — APPROVED** [review]

**Comment by JStickler** [comment]
FYI - Jaeger 1.15 was released on November 7 (https://github.com/jaegertracing/jaeger/releases/tag/v1.15.0) so should this be merged?

**Comment by objectiser** [comment]
Yes this can be merged, there is the upstream 1.15 (and now 1.16) release - and there is also [product image builds](https://issues.redhat.com/browse/TRACING-844) for 1.15.

**Pull request merged**